Support for ST7567 LCD I2C display driver

geeksville opened the pull request:
This is a WIP driver for the ST7567 LCD chip.  Do not merge.  I'm just opening this as a draft to remind me and come back and make a clean PR later.

Thanks for the lib.  I used it on previous versions of our [meshtastic](https://github.com/meshtastic/Meshtastic-device) project, and just now added a device that has an LCD screen.

geeksville committed:
Fix problem of transitions getting faster with each call of setTargetFPS

    On my app I noticed that each time I called setTargetFPS the transitions
    would become faster and faster, until finally they don't even take one
    frame.  The problem was that the FP math for calculating changeRatio
    would (because of rounding err) keep shrinking the number of ticksPerFrame.

    This change instead computs ticksPerFrame/perTransition based on the
    requested display time, so error can't accumulate from a series of calls.

    (In my app I only bump up target FPS before each transition but then
    I lower it down again afterwards to save CPU)

geeksville committed:
Allow this library to work on other Arduinos (NRF52) besides ESPs

geeksville committed:
Merge remote-tracking branch 'root/master'

geeksville committed:
Let users specify -1 for scl/sda to mean 'use current value'
(Also makes it easier to share code when users of library are targeting
multiple platforms)

geeksville committed:
fix missing dependency, std::max is defined in algorithm

mrvdb committed:
Treat SH1106 controller the same as SSD1306

- allow -1 as value for _sda and _scl parameters
  (see:
  https://github.com/meshtastic/esp8266-oled-ssd1306/commit/6a56151888f3b0ca904a918091e77dd903fb1432
  )
- use same autoinit logic

geeksville committed:
Merge pull request #1 from mrvdb/ssh1106-support

Treat SH1106 controller the same as SSD1306

geeksville committed:
make sh1106 displays work on NRF52

geeksville committed:
Super ugly early version of a ST7567 LCD driver

geeksville committed:
add basic contrast support for the LCD

crossan007 committed:
fix what appears to be a bug in the configurable distance between UI frame "dots"

crossan007 committed:
add more information to the OLEDDisplayUiState about the transition

geeksville committed:
Merge pull request #2 from crossan007/master

Bugfix and Improve the details in OLEDDisplayUiState during transition

crossan007 committed:
add frame notification bounceyness

crossan007 committed:
tweaks for debugging?

crossan007 committed:
use std::vector instead of junky arrays

crossan007 committed:
mostly working panel notifications

crossan007 committed:
fix header

crossan007 committed:
Cleanup the frame notification API so we're not passing ugly pointers

crossan007 committed:
prevent adding a notification to the currently active frame

crossan007 committed:
Nicely working callback to inform of currently active screen

crossan007 committed:
return -1 in getFirstNotifyingFrame if there are no notifications

geeksville committed:
Merge pull request #3 from crossan007/adjust_frame_notifications

add frame notification bounceyness

mc-hamster committed:
Merge branch 'master' of https://github.com/ThingPulse/esp8266-oled-ssd1306 into ThingPulse-master

mc-hamster committed:
Merge branch 'ThingPulse-master' into update-from-head

mc-hamster committed:
Fix typo /Drawen/Drawn/

mc-hamster committed:
update /Drawn/Drawen/

mc-hamster committed:
update Drawen to Drawn

mc-hamster committed:
More drawen to drawn

mc-hamster committed:
Merge pull request #5 from meshtastic/update-from-head

Update from head

caveman99 committed:
Factor actual fontdata out to a cpp file so the font won't get embedded in every namespace that includes the .h file.

mkinney committed:
Merge pull request #6 from caveman99/master

Factor actual fontdata out to a cpp file...

mkinney committed:
adding library for caveman99

mkinney committed:
Merge pull request #7 from mkinney/add_lib_for_caveman

adding library for caveman99

caveman99 committed:
Another Tryfix for Github CI

caveman99 committed:
Update OLEDDisplayUi.h

mkinney committed:
Merge pull request #8 from caveman99/master

Fix remaining CI problems on native build

sachaw commented:
@marcelstoer Hey, any chance we could get these fixes merged, we're currently trying to minimize the amount of out-of-sync forks we have, thanks.

marcelstoer commented:
We ignore PRs labelled as 'Draft' by the authors. We'll take a look once the author considers it done.

mc-hamster commented:
@sachaw > Hey, any chance we could get these fixes merged, we're currently trying to minimize the amount of out-of-sync forks we have, thanks.

If memory serves, this change has an issue with one of Meshtastic build targets. That's why it's still in draft.

caveman99 committed:
Helper for Autodetect Code

Include a different header to select the OLED Controller based on a parameter that is probed early on the I2C bus in the Meshtastic Firmware

caveman99 committed:
Merge branch 'ThingPulse:master' into master

mc-hamster committed:
Merge branch 'ThingPulse:master' into master

caveman99 committed:
Merge branch 'ThingPulse:master' into master

caveman99 committed:
add support for SH1107 variant on 128x128

caveman99 committed:
shift for auto detect too

caveman99 committed:
Test: SH1107 Fix

dizcza committed:
fixed name collision for keyword NONE

Mictronics committed:
Enable selection of I2C wire interface on Pico RP2040 platform.

Mictronics committed:
Typo fixed.

thebentern committed:
Merge pull request #10 from Mictronics/master

Enable selection of I2C wire interface on Pico RP2040 platform.

caveman99 committed:
Merge pull request #11 from ThingPulse/master

upstream changes

caveman99 committed:
Move SetGeometry to public space

caveman99 committed:
Merge pull request #12 from meshtastic/caveman99-smallscreen

Move SetGeometry to public space

caveman99 committed:
Merge branch 'ThingPulse:master' into master

thebentern committed:
Always break on newline character

thebentern committed:
Merge pull request #13 from meshtastic/newline

Always break on newline character

todd-herbert committed:
Restore log buffer: revert upstream changes

todd-herbert committed:
Leave the GitHub runner alone

caveman99 committed:
Merge pull request #14 from todd-herbert/master

Restore log buffer: revert upstream changes

caveman99 committed:
speed up code when autodetected too.

jp-bennett committed:
Make the FrameCallback a std::function to make lambdas easier

jp-bennett committed:
Merge pull request #9 from dizcza/master

fixed name collision for keyword NONE

alexbegoon committed:
Fix wrong I2C number detection for ESP32C6 target (#16)

marcelstoer commented:
@caveman99 @geeksville @mc-hamster @sachaw (tagging meshtastic folks)

Do you still plan on getting this out of draft status at some point? GitHub reminds me every time a commit is added to your master branch 😜 We'd still be open to get this merged to add ST7567 support.

rallebeckfisch commented:
Hi, it seams the display has half screen , not 128x128 , i think 128x64

caveman99 committed:
fix sh1107 on nrf52 (#18)

caveman99 committed:
align the buffers to 8 bit vertical resolutions for 'crooked' displays. (#19)

Welp, Yolo!

jp-bennett committed:
Add horizontal offset for ssd1306spi (#20)

jp-bennett committed:
Add delay_poweroff option

vidplace7 committed:
Preserve base class 'Print' overloads (address warnings)

thebentern committed:
Merge pull request #22 from vidplace7/arduino-3.x

Preserve base class 'Print' overloads (address warnings)

jp-bennett committed:
Change data types from int16_t to int32_t (#23)

On a large display, this was overflowing the int16_t variables, and causing screen draw problems.
